docs/versioned_docs/version-2.1/20-usage/20-workflow-syntax.md

```diff
@@ -803,6 +803,17 @@ when:
   event: deployment
 ```
 
+### `task`
+
+Execute a step for deployment events matching the target deployment task:
+
+```diff
+when:
+  task: migration
+  event: deployment
+```
```

Comment:
Not sure if we really need a separate when element here. To not bloat the list I think it would be better to just use `evaluate: 'CI_PREV_PIPELINE_DEPLOY_TASK == "my-task"'` instead.

Reply:
Would still prefer the environment variable version as this is a quite specific feature for github only and that way it wont be to present in the docs / yaml autocompletion etc. But I wont block it and leave that up to you.

Reply:
> I'm totally fine with not adding a specific yaml key, but then I'd say we should also remove environment for consistency reasons there.

True. That would be breaking, so we would need to deprecate it first?